app/src/main/java/com/github/se/polyfit/viewmodel/meal/OverviewViewModel.kt

```diff
@@ -165,4 +169,12 @@ constructor(private val mealDao: MealDao, private val spoonacularApiCaller: Spoo
     // Return the meal with the lowest calories
     return lowestCalorieMeal
   }
+
+  fun getUserName(): String {
+    return if (user.displayName != null) {
+      user.displayName!!
+    } else {
+      user.email
+    }
+  }
```

Comment:
From what I understood using `when` instead of if statements would lower the complexity
thus you could do something like

```
fun getUserName(): String {
    return when {
        user.displayName != null -> user.displayName!!
        else -> user.email
    }
}
```

Reply:
So it's like a switch? Awesome, I didn't even know that `when` existed. Thank you for letting me know. Yeah, that's a great idea to use one here.

Reply:
Yeah if I remember correctly that how the switch syntax in kotlin is, quite different from other languages